tests/pytest/test__run_on_whole_repo.py

```diff
@@ -0,0 +1,108 @@
+from pathlib import Path
+
+import pytest
+import yaml
+
+from pre_commit_terraform._run_on_whole_repo import is_function_defined
+from pre_commit_terraform._run_on_whole_repo import is_hook_run_on_whole_repo
+
+
+# ?
+# ? is_function_defined
+# ?
+def test_is_function_defined_existing_function():
+    def sample_function():
+        pass
+
+    scope = globals()
+    scope['sample_function'] = sample_function
```

Comment:
But really.. Can't you just not go the unnecessarily complicated route? KISS.
```suggestion
    scope = {'sample_function': sample_function}
```